src/main/java/bio/terra/service/filedata/flight/ingest/IngestPopulateFileStateFromFileStep.java

```diff
@@ -19,51 +12,37 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
-import com.google.cloud.storage.Storage;
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
 
 // Populate the files to be loaded from the incoming array
-public class IngestPopulateFileStateFromFileStep implements Step {
+public abstract class IngestPopulateFileStateFromFileStep implements Step {
   private final LoadService loadService;
   private final int maxBadLines;
   private final int batchSize;
-  private final GcsPdao gcsPdao;
 
   public IngestPopulateFileStateFromFileStep(
-      LoadService loadService, int maxBadLines, int batchSize, GcsPdao gcsPdao) {
+      LoadService loadService, int maxBadLines, int batchSize) {
     this.loadService = loadService;
     this.maxBadLines = maxBadLines;
     this.batchSize = batchSize;
-    this.gcsPdao = gcsPdao;
   }
 
-  @Override
-  public StepResult doStep(FlightContext context) {
+  private ObjectMapper getObjectMapper() {
     // Ensure that file ingestion works with extra key-value pairs
-    ObjectMapper objectMapper =
-        new ObjectMapper()
-            .registerModule(new Jdk8Module())
-            .registerModule(new JavaTimeModule())
-            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-
-    FlightMap inputParameters = context.getInputParameters();
-    BulkLoadRequestModel loadRequest =
-        inputParameters.get(JobMapKeys.REQUEST.getKeyName(), BulkLoadRequestModel.class);
+    return new ObjectMapper()
+        .registerModule(new Jdk8Module())
+        .registerModule(new JavaTimeModule())
+        .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+  }
 
-    FlightMap workingMap = context.getWorkingMap();
-    UUID loadId = UUID.fromString(workingMap.get(LoadMapKeys.LOAD_ID, String.class));
-    GoogleBucketResource bucketResource =
-        FlightUtils.getContextValue(context, FileMapKeys.BUCKET_INFO, GoogleBucketResource.class);
-    Storage storage = gcsPdao.storageForBucket(bucketResource);
-    String projectId = bucketResource.projectIdForBucket();
+  public void readFile(BufferedReader reader, UUID loadId) {
```

Comment:
Is it okay to pass around the BufferedReader like this? Usually it lives inside of the try statement.

Reply:
Ah - so, I can do this, but then I would need to manually close the Buffered reader. https://www.baeldung.com/java-buffered-reader#3-closing-the-stream